sdk/azidentity/managed_identity_credential.go

```diff
@@ -95,9 +95,9 @@ func (c *ManagedIdentityCredential) GetToken(ctx context.Context, opts azcore.To
 		addGetTokenFailureLogs("Managed Identity Credential", err, true)
 		return nil, err
 	}
-	// The following code will remove the /.default suffix from any scopes passed into the method since ManagedIdentityCredentials expect a resource string instead of a scope string
-	opts.Scopes[0] = strings.TrimSuffix(opts.Scopes[0], defaultSuffix)
-	tk, err := c.client.authenticate(ctx, c.id, opts.Scopes)
+	// managed identity endpoints require an AADv1 resource (i.e. token audience), not a v2 scope, so we remove "/.default" here
+	scopes := []string{strings.TrimSuffix(opts.Scopes[0], defaultSuffix)}
```

Comment:
I think it's okay to allocate a slice here unconditionally because we'll normally need to trim the suffix but I'm curious, do we prefer to limit such allocations generally?

Reply:
It seems totally reasonable here b/c we need to mutate the input.  @jhendrixMSFT, Is there design guidance surrounding this, should there be?

Reply:
Depending on escape analysis, `scopes` might or might not be allocated on the heap.  Given that it's pretty small, I would guess that it's allocated on the stack given that it's passed to callers but doesn't escape the callee, however you'd have to profile to be sure.

Generally speaking, the language is opaque WRT when things are stack vs heap allocated as this can change as the compiler improves.  The general guidance is to write the code with the semantics you want, potentially changing it based on profiling (obviously there are exceptions but this is a good place to start).